plugin/entryBase.go

```diff
@@ -87,13 +91,42 @@ func (e *EntryBase) setID(id string) {
 	e.washID = id
 }
 
-func (e *EntryBase) getTTLOf(op actionOpCode) time.Duration {
+func (e *EntryBase) getTTLOf(op defaultOpCode) time.Duration {
 	return e.ttl[op]
 }
 
 // OTHER METHODS USED TO FACILITATE PLUGIN DEVELOPMENT
 // AND TESTING
 
+// Name returns the entry's name as it was passed into
+// plugin.NewEntry. You should use e.Name() when making
+// the appropriate API calls within your plugin.
+func (e *EntryBase) Name() string {
+	return e.name()
+}
+
+// SetInitialAttributes sets the entry's initial attributes. Use it
+// after creating the entry via a call to NewEntry.
+func (e *EntryBase) SetInitialAttributes(attr EntryAttributes) {
```

Comment:
I chose not to cache `meta` b/c I'm thinking that hundreds of thousands of lock acquisitions in `List` could slow things down. Happy to cache if that isn't an issue, although that could introduce other problems (e.g. how do you know this `meta`'s the freshest [if we just override it]?).